### PR TITLE
Opacity scale

### PR DESCRIPTION
**Pull request by remydenton**
## Jira

http://vjira2:8080/browse/BDS-615

## Summary

Adds a new opacity scale setting to standardize opacities

## Details

The first three commits break the new work down into the three parts that make up the new opacity scale (new setting, new tools, and new utilities).  The remaining commits update documentation and usage.

Questions/comments:
- Should the utility classes be formatted with `--`?  There is inconsistency in the existing ones (e.g. `.u-bolt-text-align-left` and `u-bolt-text-decoration-underline` vs. `u-bolt-shadow--level-10` and `u-bolt-z-index--tooltip`)
- Does the way this PR handles the deprecation of the existing (most likely unused) utility classes make sense?
- Most of the opacities I updated were already in the new scale (0, .2, .4, .6, .8, 1).  However, I replaced the existing link opacities of .7 (hover) and .5 (active) with the closest values from this new scale (.8 and .6 respectively).  My rationale was that the difference should be hardly noticeable and the consistency is an improvement.
- Furthermore, there are a number of existing places, especially those using rgba(), that use various opacities that don't appear in the new scale.  Should we:
  - Update these to fit the new scale?
  - Expand the new scale to include more values?
  - Accept that exceptions like this are to be expected?

## How to test
- Review the utility class demo at https://feature-opacity-scale.boltdesignsystem.com/pattern-lab/?p=styleguide-opacity
- Confirm parity for existing patterns that have been updated to use the new opacity scale, such as Share (https://feature-opacity-scale.boltdesignsystem.com//pattern-lab/?p=viewall-components-share)

## Deprecations
Three opacity-related utility classes are deprecated and will be removed in the next major Bolt release.  Replace them with the recommended classes shown in the table below or one of the other classes documented at https://feature-opacity-scale.boltdesignsystem.com/pattern-lab/?p=styleguide-opacity

| Deprecated utility classes | Suggested replacement
| --------------------- | -----
| `.u-bolt-opacity-75` |`.u-bolt-opacity-80`
| `.u-bolt-opacity-50` |`.u-bolt-opacity-60`
| `.u-bolt-opacity-25` |`.u-bolt-opacity-20`

### CODE REVIEW

**Comment by mikemai2awesome** [comment]
@remydenton 

We had a rule to have no double dash in utility name, because utilities are very specific in what CSS property it changes and what value it gives, it does not require the BEM structure. I don't think we documented this though.

File: packages/global/styles/07-utilities/_utilities-opacity.scss [hunk 1]
```diff
@@ -1,28 +1,49 @@
...
 
 @import '@bolt/core';
 
-$bolt-opacity-props: (0, .25, .50, .75, 1);
-
 @mixin bolt-opacity-utils($breakpoint: null) {
-  @each $value in $bolt-opacity-props {
-    $name: $value * 100;
+  @each $name, $value in $bolt-opacities {
     .u-bolt-opacity-#{$name}#{$breakpoint} {
```
Comment (mikemai2awesome) [inline packages/global/styles/07-utilities/_utilities-opacity.scss]: Can we simplify this by doing this?

```scss
@each $opacity in $bolt-opacities {
  .u-bolt-opacity-#{$opacity} {
    @include bolt-opacity($opacity);
  }

  @each $breakpoint in $bolt-breakpoints {
    $breakpoint-name: nth($breakpoint, 1);

    .u-bolt-opacity-#{$opacity}\@#{$breakpoint-name} {
      @include bolt-mq($breakpoint-name) {
        @include bolt-opacity($opacity);
      }
    }
  }
}
```

Comment (remydenton) [reply packages/global/styles/07-utilities/_utilities-opacity.scss]: That does seem better-- I will update it.

File: apps/pattern-lab/src/_patterns/01-styleguide/40-utilities/opacity.twig [hunk 2]
```diff
@@ -8,17 +8,17 @@
     <p>Add the utility class <code>.u-bolt-opacity-xx</code> where xx equals the opacity you wish to apply.</p>
     <p><strong><u>Possible values include:</u></strong></p>
     <ul>
-      {% for opacity in opacities %}
-        <li><code>{{ opacity * 100 }}</code></li>
+      {% for key, value in opacities %}
```
Comment (sghoweri) [inline apps/pattern-lab/src/_patterns/01-styleguide/40-utilities/opacity.twig]: 👍

**Pull request merged**